### ISSUE TITLE
[Gate 02] Foundation-first #69~#78 cumulative 100% closure

### ISSUE THREAD

**Issue by bonggyulim** [solar-ai-dev/mcp-work-agent#97]
## Fixed branch / dependency
Validate only `refactor/canonical-architecture-migration`.

Do not start until #81 and every implementation Wave #98~#103 are complete and pushed.

```text
#81 → #98 → #99 → #100 → #101 → #102 → #103 → #97
```

Issues #69~#80 and #82~#96 are historical evidence only. They are not implementation authority after the foundation-first reset.

No successor migration starts until this Gate returns **100% PASS at one pushed HEAD**.

## Mission
Independently validate the actual final repository result against synchronized Canonical Source and the exact Wave contracts.

Priority of evidence:

```text
actual final repository files/runtime/tests
+ synchronized Canonical Source
> Wave completion reports
> historical validator/coding-agent reports
```

A prior finding is not automatically true at final HEAD. Re-prove it from code. A prior “gap” that was already fixed before the Waves must not trigger duplicate implementation.

## Validation refs — record actual SHAs
At execution time record:

- cumulative pre-migration base: `379352bedcfd346c0a7e4971f2b70adb370f85b7`;
- investigated pre-source-sync implementation baseline: `2f344e4b93ace4d36a30fd0b6a8d6b12f1036c9c`;
- #81 source-sync SHA;
- #98, #99, #100, #101, #102, #103 pushed SHAs;
- final validation HEAD = exact #103 pushed SHA.

Never substitute reported SHAs for repository objects.

## Authority
Read final HEAD:

`00 Project Source Guide → applicable 01~15 concern owner → 04-A/04 where applicable → 16 Source + 16/00..13 → latest Ledger → Wave #98~#103 → current code/tests`.

The Ledger is a closed implementation index; it did not inspect production code or assign dispositions. Final validation must verify both canonical correctness and preservation-first Phase-2 evidence.

## Preservation-first validation rule
For every migrated responsibility verify the actual applied path was defensible under:

```text
KEEP
→ MOVE / RENAME / MOVE_RENAME
→ SPLIT / MERGE
→ targeted correction
→ REWRITE only when semantic/crash/security mismatch made preservation unsafe
→ DELETE only after caller cut-over
→ CREATE only when no reusable implementation existed
```

BLOCK if repository/history evidence shows:
- canonical-compatible logic/test/fixture/SQL/runtime/UI/Adapter code was discarded without responsibility-level justification;
- path/name mismatch alone caused equivalent delete+recreate;
- a broad implementation was deleted despite reusable responsibilities that should have been split/moved;
- an old authority/wrapper was deleted before callers/useful implementation were migrated;
- a genuinely unmigrated capability was deleted only to satisfy a negative scan;
- REWRITE/DELETE/CREATE lacks evidence.

Preservation does **not** permit permanent duplicate authority or compatibility chains.

## Verified baseline facts the final Gate must use as regression anchors
These were directly observed at `2f344e4b...`. At final HEAD verify the intended disposition/result rather than blindly re-raising old findings.

### Baseline structures that were already correct and should normally survive
- `api/schemas/runs/start_run.py` already accepted `selected_resource_handles` only;
- `api/routes/runs.py::start_run()` already resolved handles before StartRun and scheduled after handler return;
- `domain/run/transitions/begin_planning.py` + guard already contained the core Context Adjustment `WAITING_APPROVAL` guard skeleton;
- `adapters/langgraph/registry/node_registry.py` already contained the 35 runtime node/profile mapping;
- `adapters/langgraph/registry/resume_target_registry.py` already had the exact safe Main stage set and excluded `ACTION_EXECUTION`;
- `CheckpointEffectiveBindingResolver` already delegated target legality to the registry;
- `api/composition.py` already injected `RequireRecoveryHandler` into Redrive and built the canonical execution/reconciliation skeleton;
- `WorkflowHandoffReconciliationLoop` was already a thin driver;
- `application/workflows/**` was already absent.

Unnecessary destruction/reimplementation of these is a preservation finding unless required by a proven new Source consequence.

### Baseline concrete defects expected to be closed
Validate final closure, not issue prose:

1. **Domain ownership/duplicates**
   - recovery lifecycle transition/guard ownership moved from `domain/run/**` to canonical recovery owner with exact reason matrix/NO_PROGRESS;
   - duplicate `domain/recovery/transitions/prepare_write_retry.py` removed after preserving/using stronger Action-owned implementation;
   - misplaced recovery/execution-attempt transition responsibilities have one owner.

2. **Persistence/UoW**
   - `adapters/persistence/unit_of_work.py → SQLiteUnitOfWork` preservation-first moved/renamed to canonical SQLite UoW path/symbol;
   - broad `adapters/persistence/repositories.py` responsibilities migrated without throwing away stronger SQL/CAS logic;
   - RecoveryContext delete/recreate ABA and bounded-read/currentness gaps closed;
   - existing 0001..0011 history unchanged; new schema fix uses forward migration only;
   - WorkflowHandoff existing CAS/retirement mechanics preserved and extended rather than replaced.

3. **StartRun/Application**
   - initial WorkflowBinding materialized before Domain guard and same-UoW durable set exact;
   - Application direct `sqlite3` ownership removed;
   - ResourceRef persists exact Registry `resource_type`, not legacy THREAD/MESSAGE/EVENT mapping;
   - exact Audit event `RUN_STARTED`;
   - post-commit scheduler behavior preserved.

4. **Scheduler/Redrive**
   - ScheduleRunExecution centrally rejects fresh obsolete NORMAL work under REAUTH/RECOVERY/CANCEL/terminal authority;
   - stale persisted admission is retired/cleared before early return;
   - active CONSUMED lineage precedes BLOCKED/PENDING;
   - fair bounded passes cannot starve later Runs;
   - durable-progress/has-more/NO_PROGRESS result semantics drive startup safely;
   - RequireRecovery is mandatory in the production-capable Redrive path;
   - do not require reimplementation of registry delegation that was already correct.

5. **Recovery Application authority**
   - canonical RequireRecoveryHandler and ResolveRecoveryHandler are the only semantic writers;
   - `ResumeRunHandler` no longer directly performs recovery lifecycle/context/audit mutation;
   - `/resolve-recovery` no longer uses legacy `ResolveMismatchRecoveryHandler` as semantic owner;
   - `/resume(RECOVERY_RECHECK)` and `/resolve-recovery` converge on canonical ResolveRecovery;
   - full persisted RecoveryContext/reason matrix/NO_PROGRESS/cleanup/audit/handoff semantics close.

6. **Planning/Review**
   - BeginPlanning no longer accepts Plan `ACTIVE` as current approval-gated Write Plan;
   - Context Adjustment target is issued/validated by ResumeTargetRegistry, not directly constructed;
   - `plan.record_review_result` exists as one canonical freshness writer or an existing equivalent was moved/merged to that owner.

7. **One-shot control/runtime**
   - BackgroundRunExecutor durably applies one-shot control to runnable checkpoint/native state before settlement clears handoff payload;
   - crash after control durability/before owner invocation loses no user decision;
   - consumed-continuation reinjects original control zero times;
   - NodeRegistry/ResumeTargetRegistry remain singular.

8. **Startup/runtime cut-over**
   - FastAPI startup is ordered, not concurrent generic startup callbacks that race canonical drains;
   - ReconcileInflightExecutions → initial Redrive drain → live loop → READY order proven;
   - LocalRunCoordinator no longer owns migrated START/resume/recovery/cancel runtime semantics;
   - concrete construction is centralized in `api/composition.py` except registered exceptions.

9. **Frontend/Connector/Security**
   - StartRun opaque selection-handle boundary preserved;
   - noncanonical frontend feature ownership migrated by move/split/merge, not wholesale UI replacement;
   - Calendar Month View separate from generic 90-day Upcoming;
   - Browser/Provider/Retrieval/local cursor continuation lifetimes separate;
   - `google_oauth_client_secret` and `GOOGLE_OAUTH_CLIENT_SECRET` authority = 0;
   - broad Connector/MCP files split/cut over while preserving provider/tool mechanics;
   - live callers use exact canonical Port boundaries.

10. **Repository roots/manifests**
   - top-level `experiments/` = 0 with its candidate/dataset artifacts preserved under canonical `evaluation/` ownership;
   - exact 43 Agent operations from 16/01, including the two mandatory deterministic rows;
   - exact 24 Port↔Adapter pairs from 16/07, OperationalCommandReplay exactly once;
   - 35 runtime-node adapter identity remains a separate manifest, not confused with 43 Agent repository operations;
   - `DiagnosticsPort` only; `DiagnosticBundlePort` authority = 0.

11. **False-positive guard**
   - do not require an invented pre-admission `PENDING → BLOCKED_BINDING` transition;
   - do not create duplicate RecoveryContext types only for physical placement;
   - do not rebuild private target tables already absent;
   - do not rename 06 runtime stage IDs to match Python function names.

## Five independent validation lanes
Run independently before final integration:

1. **Canonical Source / behavior / spec-to-code / preservation**
2. **Transaction / crash / replay / handoff / recovery safety**
3. **Single authority / caller / import / export / compatibility / dependency closure**
4. **Interface / frontend / Connector / continuation / security / OAuth / operational replay**
5. **Test / migration / regression / exact-manifest / preservation evidence**

Do not share findings between lanes until each lane finishes. Final integrator deduplicates root findings.

## Mandatory crash matrix
At least test/trace:

1. every StartRun UoW participant before commit;
2. StartRun commit → crash before schedule;
3. admission claim → crash before worker;
4. one-shot control checkpoint commit → crash before settlement;
5. settlement/control durability → crash before owner invocation;
6. restart after 1 and after 2+ descendant checkpoints;
7. Run preempted by Reauth/Recovery/Cancel/terminal before fresh NORMAL claim;
8. Run version changes after claim before settlement;
9. BLOCKED_BINDING before RequireRecovery;
10. RecoveryContext commit before handoff supersession;
11. stable no-progress row at front of full batch;
12. historical CONSUMED H1 plus active H2;
13. ResolveRecovery failure before context settlement/audit/handoff/commit;
14. RecoveryContext clear/recreate ABA;
15. committed PENDING through startup/live lifecycle.

Unsafe duplicate execution, lost control/work, stale owner I/O, starvation, orphan facts and unbounded loops must all be zero.

## Exact structural/caller proof
For every affected capability record:

```text
capability / Ledger row
canonical authority
original implementation source
disposition/preservation evidence
expected callers
observed canonical callers
old callers
old imports/exports
compat/wrapper paths
duplicate writer count
canonical test owner
targeted test result
unresolved debt classification
```

A canonical file existing without intended caller cut-over is not PASS.

## Finding classification
Exactly one:

`BLOCK | NON_BLOCK | FALSE_POSITIVE | DOCUMENTATION_BLOCK | DEFER_TO_UNMIGRATED_CAPABILITY | ALREADY_CLOSED`.

A genuinely unmigrated capability may be deferred only if its exact current owner/caller is identified and it is not a second authority/bypass for the migrated scope.

## Execution evidence
At exact final HEAD:
- run all available targeted/unit/contract/integration/component/frontend/runtime/security/migration/architecture suites required by #98~#103;
- run `git diff --check` and applicable Ruff/mypy/typecheck/build;
- compare any full-suite failure with the exact comparison base using identical commands;
- record commands, exit status, test counts and failures;
- report `NOT_EXECUTED` when not run;
- never infer PASS from file presence or Wave reports.

## 100% PASS condition
All at one pushed HEAD:

- Canonical behavior PASS;
- exact path/file/symbol/test mapping PASS;
- intended caller cut-over PASS;
- old callers/imports/concrete exports = 0;
- duplicate live authorities = 0;
- forbidden compatibility/wrapper/barrel paths = 0;
- forbidden dependency/composition edges = 0;
- unsafe crash cuts = 0;
- stale owner I/O paths = 0;
- lost durable control/work paths = 0;
- transaction/orphan fact gaps = 0;
- Recovery authority/context/NO_PROGRESS gaps = 0;
- security/trust-boundary bypasses = 0;
- continuation/token/credential leaks = 0;
- client-secret authority = 0;
- applied migration history immutable and populated upgrades PASS;
- exact 43/24/35 manifest separation/enforcement PASS;
- top-level repository root closure PASS;
- strong behavioral tests with no regression hiding;
- unjustified canonical-compatible implementation/test/fixture loss = 0;
- unjustified equivalent DELETE+CREATE churn = 0;
- responsibility-level REWRITE/DELETE/CREATE evidence gaps = 0;
- Documentation Blocker = 0 for migrated scope.

## Final output
Produce:

- overall `PASS | BLOCKED | BLOCKED_BY_DOCUMENTATION_GAP`;
- per-Wave verdict #81 and #98~#103;
- capability-family verdict for historical #69~#78 scope;
- deduplicated findings with exact Source/Ledger/path/symbol/evidence/minimal owner correction;
- preservation/disposition evidence summary;
- crash matrix and closure metrics;
- exact test commands/results and baseline comparisons;
- exact final HEAD;
- successor-work decision.

## Stop condition
If anything is not 100% closed, do not start successor work. Create only bounded correction issues assigned to the existing semantic-owner Wave boundary, complete them, push, and rerun this Gate at the new single HEAD.

**Comment by bonggyulim** [comment]
Superseded by the execution-order reset in #104. The old cumulative Wave gate is not executable while Domain remains incomplete. A new cumulative/final validation gate should be created only after each layer is closed in the new order.
